externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/FlinkSQLEngine.scala

```diff
@@ -72,6 +75,20 @@ object FlinkSQLEngine extends Logging {
     try {
       val flinkConfDir = CliFrontend.getConfigurationDirectoryFromEnv
       val flinkConf = GlobalConfiguration.loadConfiguration(flinkConfDir)
+
+      val executionTarget = flinkConf.getString(DeploymentOptions.TARGET)
+      // set cluster name for per-job and application mode
+      executionTarget match {
+        case "yarn-per-job" | "yarn-application" =>
```

Comment:
Will we consider `yarn-session`?

Reply:
IIUC, in session mode we need to start a yarn session beforehand, so the cluster name/id is already determined, so here only includes application-mode cluster and per-job cluster, which is initialized lazily.

Reply:
yes, both for session and yarn-session are pre-started.